### ISSUE TITLE
Terraform v0.13.0 Invalid for_each when using aws_iam_policy_document data source

### ISSUE THREAD

**Issue by adamhorden** [hashicorp/terraform#25665]
**Terraform Version**

```
> terraform -v

Terraform v0.13.0-rc1
```

I am not sure if this bug report is a provider issue or a core Terraform issue.

We make heavy use of `aws_iam_policy_document`. 
As this is just generating JSON in many of the use cases in our infrastructure, we started to see many:

**Error: Invalid for_each argument**

After some digging if we do:

```
locals {
  value = data.aws_iam_policy_document.this_he_aws_tf_mod_aws_iam_rle_amz_ddb_as.json
}
```

We get a error. We pass this into an IAM module.

As this is just generating JSON why is it waiting till apply to generate this JSON? I assumed at a plan stage this would just return the JSON.

Surprisingly with all my previous testing I had not come across this before in v0.12 or when I was testing v0.13 from git.

Example:

```
  # (config refers to values not yet known)                                                                                                                                                                       
 <= data "aws_iam_policy_document" "this_he_aws_tf_mod_aws_iam_rle_amz_ddb_as"  {                                                                                                                                 
      + id   = (known after apply)                                                                                                                                                                                
      + json = (known after apply)                                                                                                                                                                                
                                                                                                                                                                                                                  
      + statement {                                                                                                                                                                                               
          + actions = [                                                                                                                                                                                           
              + "sts:AssumeRole",                                                                                                                                                                                 
            ]                                                                                                                                                                                                     
          + effect  = "Allow"                                                                                                                                                                                     
          + sid     = "HeAwsTfAppItfSrvStkBeHeAwsTfModAmzDdbAsEnv1"                                                                                                                                               
                                                                                                                                                                                                                  
          + principals {                                                                                                                                                                                          
              + identifiers = [                                                                                                                                                                                   
                  + "application-autoscaling.amazonaws.com",                                                                                                                                                      
                ]                                                                                                                                                                                                 
              + type        = "Service"                                                                                                                                                                           
            }                                                                                                                                                                                                     
        }                                                                                                                                                                                                         
    } 
```

I can always generate the JSON and use `file` but wanted to make sure I had not come across a bug :bug: first before refactoring.

Thanks for all of the hard work that has gone into Terraform v0.13 :grimacing:.

**Comment by jbardin** [comment]
Hi @adamhorden,

The plan is indicating that there is something causing the evaluation of the `aws_iam_policy_document` to be deferred, but we can't discern why this is happening here without the configuration. Do you have an example you can show which demonstrates this?

**Comment by jbardin** [comment]
I'm going to close this out since it has no reproduction case.
Chances are it is one of the known cases that cause a data source to be deferred until apply, which may have already been solved, or will be resolved with 0.14.

Thanks!
